src/ncp/example_vendor_hook.cpp

```diff
@@ -77,7 +77,7 @@ otError NcpBase::VendorGetPropertyHandler(spinel_prop_key_t aPropKey)
         // TODO: Implement your property get handlers here.
         //
         // Get handler should retrieve the property value and then encode and write the
-        // value into the NCP buffer. If the "get" operation itself fails, handler should
+        // value into the NCP buffer. If the "get" operation itself fails, `NcpBase` will
```

Comment:
@bertoldi-silabs the model used here is intentional to allow for vendor specific errors (and/or new mechasnim to report error). Suggest not changing this.

Same model is used for internal or vendor "get handlers":
https://github.com/openthread/openthread/blob/6f909a1d6a9213cb137615af205aa29e1aca39b1/src/ncp/ncp_base.hpp#L393-L406

Reply:
@abtink Thanks for the feedback. I think that sounds like an interesting idea in the case for which new mechanisms may be considered for reporting errors. 

The question I still have is how can a vendor "get handler" write a `LAST_STATUS` without access to the spinel header? If my handler returns OT_ERROR_FAILED, it is unable to call `WriteStatusLastFrame` and the base class also will not handle it.

Reply:
> We provide `OverwriteWithLastStatusError()` for this.

Adding this in my handler worked without the need for any of the changes in this PR.
I will close this PR now. Thank you for the help!